rust/kernel/task.rs

```diff
@@ -180,13 +180,22 @@ impl Task {
 
         // SAFETY: Since the kthread creation succeeded and we haven't run it yet, we know the task
         // is valid.
-        let task = unsafe { &*(ktask as *const Task) }.into();
+        let task: ARef<_> = unsafe { &*(ktask as *const Task) }.into();
+
+        // Wakes up the thread, otherwise it won't run.
+        task.wake_up();
```

Comment:
FWIW: this is going to change to a call to `wake_up_new_task`.

@fbq have you seen this? https://lore.kernel.org/lkml/871qvbutex.fsf_-_@email.froward.int.ebiederm.org/

This affects the design for the generic task changes you have. I think we should use type states to decide which methods are available.

Reply:
Thanks for the pointer, I haven't seen that before.